File: src/main/java/me/diax/diax/music/AudioPlayerSendHandler.java
package me.diax.diax.music;

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer;
import com.sedmelluq.discord.lavaplayer.track.playback.AudioFrame;
import net.dv8tion.jda.core.audio.AudioSendHandler;

public class AudioPlayerSendHandler implements AudioSendHandler {

    private final AudioPlayer player;
    private AudioFrame frame;

    public AudioPlayerSendHandler(AudioPlayer player) {
        this.player = player;
    }

    @Override
    public boolean canProvide() {
        if (frame == null) frame = player.provide();
        return frame != null;
    }

    @Override
    public byte[] provide20MsAudio() {
        if (frame == null) frame = player.provide();
        byte[] data = frame != null ? frame.data : null;
        frame = null;
        return data;
    }

    @Override
    public boolean isOpus() {
        return true;
    }
}

File: src/main/kotlin/me/diax/diax/music/AudioPlayerSendHandler.kt
package me.diax.diax.music

import com.sedmelluq.discord.lavaplayer.player.AudioPlayer
import com.sedmelluq.discord.lavaplayer.track.playback.AudioFrame
import net.dv8tion.jda.core.audio.AudioSendHandler

class AudioPlayerSendHandler(private val player: AudioPlayer) : AudioSendHandler {
    private var frame: AudioFrame? = null

    override fun canProvide(): Boolean {
        if (frame == null) frame = player.provide()
        return frame != null
    }

    override fun provide20MsAudio(): ByteArray? {
        if (frame == null) frame = player.provide()
        val data = if (frame != null) frame!!.data else null
        frame = null
        return data
    }

    override fun isOpus(): Boolean {
        return true
    }
}